tests/validators/tests.py

```diff
@@ -205,6 +205,11 @@
 
     (URLValidator(EXTENDED_SCHEMES), 'file://localhost/path', None),
     (URLValidator(EXTENDED_SCHEMES), 'git://example.com/', None),
+    (URLValidator(EXTENDED_SCHEMES), 'ssh://example.com/', None),
+    (URLValidator(EXTENDED_SCHEMES), 'git+ssh://git@github.com/example/hg-git.git', None),
+
+    # Empty schemes should not be accepted
+    (URLValidator(schemes=['']), '://www.example.com', ValidationError),
```

Comment:
protocol-relative URLs are mainly implemented by browsers. They just use whatever scheme was used by parent page(http, https or even file). So in a way they are not completely scheme-less. 
All of the RFCs I considered had same format of <scheme>:<scheme-specific-part> which suggests that `:` is compulsory.